executor/tests/test_executor.py

```diff
@@ -702,24 +682,112 @@ def assertions(driver, task_id, sandbox_directory):
+            else:
+                self.fail('{} does not exist.'.format(stdout_name))
+
+            stderr_name = ensure_directory('build/stderr.' + str(task_id))
+            if os.path.isfile(stderr_name):
+                with open(stderr_name) as f:
+                    file_contents = f.read()
+                    self.assertEqual(num_iterations * 25, file_contents.count('X'))
+            else:
+                self.fail('{} does not exist.'.format(stderr_name))
+
+        stop_signal = Event()
+        Thread(target=sleep_and_set_stop_signal, args=(stop_signal, 60)).start()
+
+        command = 'for i in `seq {}`; ' \
+                  'do printf "XXXXXXXXXXXXXXXXXXXXXXXXX"; printf "XXXXXXXXXXXXXXXXXXXXXXXXX" >&2; done; ' \
+                  'echo "Done."'.format(num_iterations)
+        self.manage_task_runner(command, assertions, stop_signal=stop_signal)
+        stop_signal.set()
+
+    def test_manage_task_long_output_multiple_lines(self):
+        num_iterations = 100000
+
+        def assertions(driver, task_id, sandbox_directory):
+
+            logging.info('Statuses: {}'.format(driver.statuses))
+            self.assertEqual(3, len(driver.statuses))
+
+            logging.info('Messages: {}'.format(driver.messages))
+            self.assertEqual(2, len(driver.messages))
+
+            actual_encoded_message_0 = driver.messages[0]
+            expected_message_0 = {'sandbox-directory': sandbox_directory, 'task-id': task_id, 'type': 'directory'}
+            assert_message(self, expected_message_0, actual_encoded_message_0)
+
+            actual_encoded_message_1 = driver.messages[1]
+            expected_message_1 = {'exit-code': 0, 'task-id': task_id}
+            assert_message(self, expected_message_1, actual_encoded_message_1)
+
+            stdout_name = ensure_directory('build/stdout.' + str(task_id))
+            if os.path.isfile(stdout_name):
+                with open(stdout_name) as f:
+                    file_contents = f.read()
+                    self.assertEqual(num_iterations * 25, file_contents.count('X'))
             else:
                 self.fail('{} does not exist.'.format(stdout_name))
 
+            stderr_name = ensure_directory('build/stderr.' + str(task_id))
+            if os.path.isfile(stderr_name):
+                with open(stderr_name) as f:
+                    file_contents = f.read()
+                    self.assertEqual(num_iterations * 25, file_contents.count('X'))
+            else:
+                self.fail('{} does not exist.'.format(stderr_name))
+
         stop_signal = Event()
+        Thread(target=sleep_and_set_stop_signal, args=(stop_signal, 60)).start()
+
+        command = 'for i in `seq {}`; ' \
+                  'do printf "XXXXXXXXXXXXXXXXXXXXXXXXX\\n"; printf "XXXXXXXXXXXXXXXXXXXXXXXXX\\n" >&2; done; ' \
+                  'echo "Done."'.format(num_iterations)
+        self.manage_task_runner(command, assertions, stop_signal=stop_signal)
+        stop_signal.set()
```

Comment:
Should we create:

```python
def test_command(command, assertions):
  stop_signal = Event()
  Thread(target=sleep_and_set_stop_signal, args=(stop_signal, 60)).start()
  self.manage_task_runner(command, assertions, stop_signal=stop_signal)
  stop_signal.set()
```

since this code is duplicated in a couple of places?

Reply:
Created `sleep_and_set_stop_signal_task` helper function.